param.go

```diff
@@ -83,6 +83,24 @@ func (paramList) ParamTypeName() string   { return "parameter list" }
 func (paramSimple) ParamTypeName() string { return "simple type" }
 func (paramObject) ParamTypeName() string { return "parameter object" }
 
+// Calls the provided function on all paramSimples in the given param tree.
+func forEachSimpleParam(param param, f func(paramSimple)) {
```

Comment:
`forEachSimpleParam` name doesn't make sense since function is not restricted to simple params

Reply:
walkSimpleParams?

Reply:
the _simple_ param is what I care about, not the first part :)

The way I see it it walks/traverses _any_ param if the type switch to be believed, not just the simple ones, right?

Reply:
Yeah, but it calls the function only on the simple params. It's a "walk the params and call this function an all simpleParams" which is why I thought forEach made sense.